tests/mock_s3_server/mock_s3_server.py

```diff
@@ -29,15 +34,42 @@ class S3Opts(Enum):
     ListParts = 6
 
 
-base_dir = os.path.dirname(os.path.realpath(__file__))
+@dataclass
+class Response:
+    path: str
+    disconnect_after_headers = False
+    generate_body_size: Optional[int] = None
+    response_json_path: str = None
+
+    def resolve_file_path(self, wrapper, request_type):
+        if self.response_json_path is None:
+            response_file = os.path.join(
+                base_dir, request_type.name, f"{self.path[1:]}.json")
+            if os.path.exists(response_file) == False:
+                wrapper.info(
+                    response_file, "not exist, using the default response")
+                response_file = os.path.join(
+                    base_dir, request_type.name, f"default.json")
+            if "throttle" in response_file:
+                # We throttle the request half the time to make sure it succeeds after a retry
+                if TrioHTTPWrapper.should_throttle is False:
+                    wrapper.info("Skipping throttling")
+                    response_file = os.path.join(
+                        base_dir, request_type.name, f"default.json")
+                else:
+                    wrapper.info("Throttling")
+                # Flip the flag
+                TrioHTTPWrapper.should_throttle = not TrioHTTPWrapper.should_throttle
+            self.response_json_path = response_file
 
 
 class TrioHTTPWrapper:
     _next_id = count()
+    retry_request_received_continuous = 0
+    should_throttle = SHOULD_THROTTLE
```

Comment:
you could probably move the throttle variable into the Response class, it's another one where we modify some global state so that something special can happen later on the next response

Reply:
Maybe I didn't get what you mean, but we need a state to control the next response, so, how can we move it to the response? We can add it to the response, but we still need a state on the server to track between requests.

Reply:
Oh, I moved it to the response as a static variable.

Reply:
we'll talk monday
i am failing to type what i mean

Reply:
talked offline and revamped the script